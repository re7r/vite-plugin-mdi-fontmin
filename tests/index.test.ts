import { afterEach, beforeEach, describe, expect, it } from 'vitest';
import { build } from 'vite';
import { fileURLToPath } from 'node:url';
import path from 'node:path';
import fs from 'node:fs';

import mdiFontmin from '../dist/index.js';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const outputDir = path.resolve(__dirname, 'output');
const inputFile = path.resolve(__dirname, 'fixtures/main.mjs');

beforeEach(() => {
  fs.rmSync(outputDir, { recursive: true, force: true });
});

afterEach(() => {
  fs.rmSync(outputDir, { recursive: true, force: true });
});

describe('vite-plugin-mdi-fontmin', () => {
  it('generates correct subset', async () => {
    await build({
      logLevel: 'silent',
      build: {
        lib: {
          entry: inputFile,
          formats: ['es'],
          fileName: () => 'bundle.js',
        },
        outDir: outputDir,
        emptyOutDir: false,
      },
      plugins: [
        mdiFontmin({
          names: ['arrow-left', 'arrow-right'],
          output: 'tests/output',
          silent: false,
          logPrefix: '[test]',
        }),
      ],
    });

    const files = fs.readdirSync(outputDir).sort();
    expect(files).toMatchSnapshot('file list matches snapshot');

    const contents: Record<string, string> = {};
    for (const file of files) {
      const ext = path.extname(file);
      const fullPath = path.join(outputDir, file);

      if (ext === '.css') {
        contents[file] = fs.readFileSync(fullPath, 'utf8');
      } else {
        contents[file] = fs.readFileSync(fullPath).toString('base64');
      }
    }

    expect(contents).toMatchSnapshot('file contents match snapshot');
  });
});
