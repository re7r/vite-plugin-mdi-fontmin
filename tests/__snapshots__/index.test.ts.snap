// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite-plugin-mdi-fontmin > generates correct subset > file contents match snapshot 1`] = `
{
  "bundle.js": "Cg==",
  "materialdesignicons-webfont.eot": "IgcAAFQGAAABAAIAAAAAAAIABQMAAAAAAAABQJABAAAAAExQAAAAABAAAAAAAAAAAAAAAAAAAAEAAAAA0AKa1gAAAAAAAAAAAAAAAAAAAAAAACoAAE0AYQB0AGUAcgBpAGEAbAAgAEQAZQBzAGkAZwBuACAASQBjAG8AbgBzAAAAAAAAFgAAVgBlAHIAcwBpAG8AbgAgADEALgAwAAAqAABNAGEAdABlAHIAaQBhAGwAIABEAGUAcwBpAGcAbgAgAEkAYwBvAG4AcwAAAAAAAQAAAAoAgAADACBPUy8yUCNrfQAAAKwAAABgY21hcAF/Az4AAAEMAAABamdseWahtUGPAAACeAAAAGBoZWFkYMlAZgAAAtgAAAA2aGhlYQNtAhgAAAMQAAAAJGhtdHgCAAAAAAADNAAAAApsb2NhABwANgAAA0AAAAAIbWF4cAEOABUAAANIAAAAIG5hbWVsc+LbAAADaAAAAqxwb3N0IFNtxwAABhQAAAA/AAQBVQGQAAUAAAFEAWYAAABHAUQBZgAAAPUAGQCEAAACAAUDAAAAAAAAAAAAABAAAAAAAAAAAAAAAFBmRWQAQABNAFQBwP/AAC4BxQBCAAAAAQAAAAAAAAAAAAAAIAAAAAAABAAAAAMAAAAkAAEAAAAAADwAAwABAAAAJAADAAoAAAFCAAQAGAAAAAIAAgAAAAD//wAA//8AAQAAAAABBgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMAAAAAAAoAAAAAAAAAAIADwBNAA8ATQAAAAEADwBUAA8AVAAAAAIAAAAAAAAAAAAAAAAAAAABAAAAAAGrAWkACAAAJRUhFwcnNxcHAav/AHUeqakeddUqdh6pqR52AAAAAAEAAAAAAagBaQAIAAA3FSEHFzcnBxdVAQB1HqmpHnXVKnYeqakedgABAAAAAQAA1poC0F8PPPUACwIAAAAAAAAAAAAAAAAAAAAAAAAAAAABqwFpAAAACAACAAAAAAAAAAEAAAHA/8AAAAIAAAAAVQGrAAEAAAAAAAAAAAAAAAAAAAACAAAAAAIAAAAAAAAAAAAABgAcADAAAQAAAAMACQABAAAAAAACAAAACgAKAAAA/wAAAAAAAAAAABAAxgABAAAAAAABABUAAAABAAAAAAACAAcAFQABAAAAAAADABUAHAABAAAAAAAEABUAMQABAAAAAAAFAAsARgABAAAAAAAGABMAUQABAAAAAAAKACsAZAABAAAAAAALABMAjwADAAEECQABACoAogADAAEECQACAA4AzAADAAEECQADACoA2gADAAEECQAEACoBBAADAAEECQAFABYBLgADAAEECQAGACYBRAADAAEECQAKAFYBagADAAEECQALACYBwE1hdGVyaWFsIERlc2lnbiBJY29uc1JlZ3VsYXJNYXRlcmlhbCBEZXNpZ24gSWNvbnNNYXRlcmlhbCBEZXNpZ24gSWNvbnNWZXJzaW9uIDEuME1hdGVyaWFsRGVzaWduSWNvbnNHZW5lcmF0ZWQgYnkgc3ZnMnR0ZiBmcm9tIEZvbnRlbGxvIHByb2plY3QuaHR0cDovL2ZvbnRlbGxvLmNvbQBNAGEAdABlAHIAaQBhAGwAIABEAGUAcwBpAGcAbgAgAEkAYwBvAG4AcwBSAGUAZwB1AGwAYQByAE0AYQB0AGUAcgBpAGEAbAAgAEQAZQBzAGkAZwBuACAASQBjAG8AbgBzAE0AYQB0AGUAcgBpAGEAbAAgAEQAZQBzAGkAZwBuACAASQBjAG8AbgBzAFYAZQByAHMAaQBvAG4AIAAxAC4AMABNAGEAdABlAHIAaQBhAGwARABlAHMAaQBnAG4ASQBjAG8AbgBzAEcAZQBuAGUAcgBhAHQAZQBkACAAYgB5ACAAcwB2AGcAMgB0AHQAZgAgAGYAcgBvAG0AIABGAG8AbgB0AGUAbABsAG8AIABwAHIAbwBqAGUAYwB0AC4AaAB0AHQAcAA6AC8ALwBmAG8AbgB0AGUAbABsAG8ALgBjAG8AbQACAAAAAAAAAAUAAAAAAAAAAAAAAAAAAAAAAAAAAAADAAMAAAECAQMKYXJyb3ctbGVmdAthcnJvdy1yaWdodAA=",
  "materialdesignicons-webfont.ttf": "AAEAAAAKAIAAAwAgT1MvMlAja30AAACsAAAAYGNtYXABfwM+AAABDAAAAWpnbHlmobVBjwAAAngAAABgaGVhZGDJQGYAAALYAAAANmhoZWEDbQIYAAADEAAAACRobXR4AgAAAAAAAzQAAAAKbG9jYQAcADYAAANAAAAACG1heHABDgAVAAADSAAAACBuYW1lbHPi2wAAA2gAAAKscG9zdCBTbccAAAYUAAAAPwAEAVUBkAAFAAABRAFmAAAARwFEAWYAAAD1ABkAhAAAAgAFAwAAAAAAAAAAAAAQAAAAAAAAAAAAAABQZkVkAEAATQBUAcD/wAAuAcUAQgAAAAEAAAAAAAAAAAAAACAAAAAAAAQAAAADAAAAJAABAAAAAAA8AAMAAQAAACQAAwAKAAABQgAEABgAAAACAAIAAAAA//8AAP//AAEAAAAAAQYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAAAAAAAKAAAAAAAAAACAA8ATQAPAE0AAAABAA8AVAAPAFQAAAACAAAAAAAAAAAAAAAAAAAAAQAAAAABqwFpAAgAACUVIRcHJzcXBwGr/wB1HqmpHnXVKnYeqakedgAAAAABAAAAAAGoAWkACAAANxUhBxc3JwcXVQEAdR6pqR511Sp2HqmpHnYAAQAAAAEAANaaAtBfDzz1AAsCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAasBaQAAAAgAAgAAAAAAAAABAAABwP/AAAACAAAAAFUBqwABAAAAAAAAAAAAAAAAAAAAAgAAAAACAAAAAAAAAAAAAAYAHAAwAAEAAAADAAkAAQAAAAAAAgAAAAoACgAAAP8AAAAAAAAAAAAQAMYAAQAAAAAAAQAVAAAAAQAAAAAAAgAHABUAAQAAAAAAAwAVABwAAQAAAAAABAAVADEAAQAAAAAABQALAEYAAQAAAAAABgATAFEAAQAAAAAACgArAGQAAQAAAAAACwATAI8AAwABBAkAAQAqAKIAAwABBAkAAgAOAMwAAwABBAkAAwAqANoAAwABBAkABAAqAQQAAwABBAkABQAWAS4AAwABBAkABgAmAUQAAwABBAkACgBWAWoAAwABBAkACwAmAcBNYXRlcmlhbCBEZXNpZ24gSWNvbnNSZWd1bGFyTWF0ZXJpYWwgRGVzaWduIEljb25zTWF0ZXJpYWwgRGVzaWduIEljb25zVmVyc2lvbiAxLjBNYXRlcmlhbERlc2lnbkljb25zR2VuZXJhdGVkIGJ5IHN2ZzJ0dGYgZnJvbSBGb250ZWxsbyBwcm9qZWN0Lmh0dHA6Ly9mb250ZWxsby5jb20ATQBhAHQAZQByAGkAYQBsACAARABlAHMAaQBnAG4AIABJAGMAbwBuAHMAUgBlAGcAdQBsAGEAcgBNAGEAdABlAHIAaQBhAGwAIABEAGUAcwBpAGcAbgAgAEkAYwBvAG4AcwBNAGEAdABlAHIAaQBhAGwAIABEAGUAcwBpAGcAbgAgAEkAYwBvAG4AcwBWAGUAcgBzAGkAbwBuACAAMQAuADAATQBhAHQAZQByAGkAYQBsAEQAZQBzAGkAZwBuAEkAYwBvAG4AcwBHAGUAbgBlAHIAYQB0AGUAZAAgAGIAeQAgAHMAdgBnADIAdAB0AGYAIABmAHIAbwBtACAARgBvAG4AdABlAGwAbABvACAAcAByAG8AagBlAGMAdAAuAGgAdAB0AHAAOgAvAC8AZgBvAG4AdABlAGwAbABvAC4AYwBvAG0AAgAAAAAAAAAFAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwADAAABAgEDCmFycm93LWxlZnQLYXJyb3ctcmlnaHQA",
  "materialdesignicons-webfont.woff": "d09GRgABAAAAAAacAAoAAAAABlQAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABPUy8yAAAA9AAAAGAAAABgUCNrfWNtYXAAAAFUAAABagAAAWoBfwM+Z2x5ZgAAAsAAAABgAAAAYKG1QY9oZWFkAAADIAAAADYAAAA2YMlAZmhoZWEAAANYAAAAJAAAACQDbQIYaG10eAAAA3wAAAAKAAAACgIAAABsb2NhAAADiAAAAAgAAAAIABwANm1heHAAAAOQAAAAIAAAACABDgAVbmFtZQAAA7AAAAKsAAACrGxz4ttwb3N0AAAGXAAAAD8AAAA/IFNtxwAEAVUBkAAFAAABRAFmAAAARwFEAWYAAAD1ABkAhAAAAgAFAwAAAAAAAAAAAAAQAAAAAAAAAAAAAABQZkVkAEAATQBUAcD/wAAuAcUAQgAAAAEAAAAAAAAAAAAAACAAAAAAAAQAAAADAAAAJAABAAAAAAA8AAMAAQAAACQAAwAKAAABQgAEABgAAAACAAIAAAAA//8AAP//AAEAAAAAAQYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAAAAAAAKAAAAAAAAAACAA8ATQAPAE0AAAABAA8AVAAPAFQAAAACAAAAAAAAAAAAAAAAAAAAAQAAAAABqwFpAAgAACUVIRcHJzcXBwGr/wB1HqmpHnXVKnYeqakedgAAAAABAAAAAAGoAWkACAAANxUhBxc3JwcXVQEAdR6pqR511Sp2HqmpHnYAAQAAAAEAANaaAtBfDzz1AAsCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAasBaQAAAAgAAgAAAAAAAAABAAABwP/AAAACAAAAAFUBqwABAAAAAAAAAAAAAAAAAAAAAgAAAAACAAAAAAAAAAAAAAYAHAAwAAEAAAADAAkAAQAAAAAAAgAAAAoACgAAAP8AAAAAAAAAAAAQAMYAAQAAAAAAAQAVAAAAAQAAAAAAAgAHABUAAQAAAAAAAwAVABwAAQAAAAAABAAVADEAAQAAAAAABQALAEYAAQAAAAAABgATAFEAAQAAAAAACgArAGQAAQAAAAAACwATAI8AAwABBAkAAQAqAKIAAwABBAkAAgAOAMwAAwABBAkAAwAqANoAAwABBAkABAAqAQQAAwABBAkABQAWAS4AAwABBAkABgAmAUQAAwABBAkACgBWAWoAAwABBAkACwAmAcBNYXRlcmlhbCBEZXNpZ24gSWNvbnNSZWd1bGFyTWF0ZXJpYWwgRGVzaWduIEljb25zTWF0ZXJpYWwgRGVzaWduIEljb25zVmVyc2lvbiAxLjBNYXRlcmlhbERlc2lnbkljb25zR2VuZXJhdGVkIGJ5IHN2ZzJ0dGYgZnJvbSBGb250ZWxsbyBwcm9qZWN0Lmh0dHA6Ly9mb250ZWxsby5jb20ATQBhAHQAZQByAGkAYQBsACAARABlAHMAaQBnAG4AIABJAGMAbwBuAHMAUgBlAGcAdQBsAGEAcgBNAGEAdABlAHIAaQBhAGwAIABEAGUAcwBpAGcAbgAgAEkAYwBvAG4AcwBNAGEAdABlAHIAaQBhAGwAIABEAGUAcwBpAGcAbgAgAEkAYwBvAG4AcwBWAGUAcgBzAGkAbwBuACAAMQAuADAATQBhAHQAZQByAGkAYQBsAEQAZQBzAGkAZwBuAEkAYwBvAG4AcwBHAGUAbgBlAHIAYQB0AGUAZAAgAGIAeQAgAHMAdgBnADIAdAB0AGYAIABmAHIAbwBtACAARgBvAG4AdABlAGwAbABvACAAcAByAG8AagBlAGMAdAAuAGgAdAB0AHAAOgAvAC8AZgBvAG4AdABlAGwAbABvAC4AYwBvAG0AAgAAAAAAAAAFAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwADAAABAgEDCmFycm93LWxlZnQLYXJyb3ctcmlnaHQA",
  "materialdesignicons-webfont.woff2": "d09GMgABAAAAAAJoAAoAAAAABkgAAAIeAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAABmAAgmoKVHABNgIkAwoLCAAEIAWFLAc/G6gFERWME7IfBW77AP2OnqNX1CdvqatrKecgHp72+3buzFsz62bJJC7ieQmNDiGLl4InIslK3PwPt/XSnlIuFDSSX70jtzEmx4SXAPifa+ql3Ql+OgCJXFKF307XDbY3sBNYJFXi3hAjgXCD7kYG8K09FHEeL5QR2/EAe+XMmy7PJGVZaGjEb3D0YOeedfvO4nPp83z8sIkA5w2CAhmj8/qyLPrbZtXEpkInkgSlUvrHC4hyOLoWTCI9abCPmIQGZ/NIIINyGNGMfphqdW6Iqqq7as8nLy6nzi7VnU1dXkyef/w9ff30/3tg8P96dKbstw4mfwoB8b68xUDw823eXjUs31RGCZb2Oyot+YO/Inbbz+OTdTIp3RMdyKrWvQnM8ZCSEGz0cyUY2nZHk1Roa025tvVCtCCXoEyNXSjX4hiqzbg3v0aL51kUVcK0lwCSer8gm/YHKegooEx7zEO58diGaheheGKqmfvtz6cqoBOVgiV6wQ3sGjL+jnxUlZsv76a90XlBBvIkWxOJQLQNGnRTq7RQ/8F/eBECA+ZIwxqycC+lCAZHPTYh6UIYyjRl9OJJQ3pyqgI6USlYohfcwK4h4++gu/ioKnemRrss7x38LE7GeZKdLWIFDN1YDBJ3XqeF+g/+w4sQGDBHGtaIh+lKEQzzo3psQtKd0IYyja3YXkvysl1/547KhN2yTKTpfFE5R99YIQtRE3aCdwEAAA==",
  "materialdesignicons.min.css": "@font-face{font-family:"Material Design Icons";src:url("./materialdesignicons-webfont.eot?v=7.4.47");src:url("./materialdesignicons-webfont.eot?#iefix&v=7.4.47") format("embedded-opentype"),url("./materialdesignicons-webfont.woff2?v=7.4.47") format("woff2"),url("./materialdesignicons-webfont.woff?v=7.4.47") format("woff"),url("./materialdesignicons-webfont.ttf?v=7.4.47") format("truetype");font-weight:normal;font-style:normal}.mdi:before,.mdi-set{display:inline-block;font:normal normal normal 24px/1 "Material Design Icons";font-size:inherit;text-rendering:auto;line-height:inherit;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}.mdi-arrow-left::before{content:"\\F004D"}.mdi-arrow-right::before{content:"\\F0054"}.mdi-blank::before{content:"\\F68C";visibility:hidden}.mdi-18px.mdi-set,.mdi-18px.mdi:before{font-size:18px}.mdi-24px.mdi-set,.mdi-24px.mdi:before{font-size:24px}.mdi-36px.mdi-set,.mdi-36px.mdi:before{font-size:36px}.mdi-48px.mdi-set,.mdi-48px.mdi:before{font-size:48px}.mdi-dark:before{color:rgba(0,0,0,0.54)}.mdi-dark.mdi-inactive:before{color:rgba(0,0,0,0.26)}.mdi-light:before{color:#fff}.mdi-light.mdi-inactive:before{color:rgba(255,255,255,0.3)}.mdi-rotate-45:before{-webkit-transform:rotate(45deg);-ms-transform:rotate(45deg);transform:rotate(45deg)}.mdi-rotate-90:before{-webkit-transform:rotate(90deg);-ms-transform:rotate(90deg);transform:rotate(90deg)}.mdi-rotate-135:before{-webkit-transform:rotate(135deg);-ms-transform:rotate(135deg);transform:rotate(135deg)}.mdi-rotate-180:before{-webkit-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg)}.mdi-rotate-225:before{-webkit-transform:rotate(225deg);-ms-transform:rotate(225deg);transform:rotate(225deg)}.mdi-rotate-270:before{-webkit-transform:rotate(270deg);-ms-transform:rotate(270deg);transform:rotate(270deg)}.mdi-rotate-315:before{-webkit-transform:rotate(315deg);-ms-transform:rotate(315deg);transform:rotate(315deg)}.mdi-flip-h:before{-webkit-transform:scaleX(-1);transform:scaleX(-1);filter:FlipH;-ms-filter:"FlipH"}.mdi-flip-v:before{-webkit-transform:scaleY(-1);transform:scaleY(-1);filter:FlipV;-ms-filter:"FlipV"}.mdi-spin:before{-webkit-animation:mdi-spin 2s infinite linear;animation:mdi-spin 2s infinite linear}@-webkit-keyframes mdi-spin{0%{-webkit-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(359deg);transform:rotate(359deg)}}@keyframes mdi-spin{0%{-webkit-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(359deg);transform:rotate(359deg)}}
",
}
`;

exports[`vite-plugin-mdi-fontmin > generates correct subset > file list matches snapshot 1`] = `
[
  "bundle.js",
  "materialdesignicons-webfont.eot",
  "materialdesignicons-webfont.ttf",
  "materialdesignicons-webfont.woff",
  "materialdesignicons-webfont.woff2",
  "materialdesignicons.min.css",
]
`;
